entity-framework/core/what-is-new/ef-core-8.0/whatsnew.md

```diff
@@ -15,770 +15,1284 @@ EF8 is available as [daily builds](https://github.com/dotnet/efcore/blob/main/do
+        {
+            modelBuilder.Entity<Customer>()
+                .ComplexProperty(e => e.Address);
+
+            modelBuilder.Entity<Order>(b =>
+            {
+                b.ComplexProperty(e => e.BillingAddress);
+                b.ComplexProperty(e => e.ShippingAddress);
+            });
+        }
 -->
-[!code-csharp[SqlQueryAllColumns](../../../../samples/core/Miscellaneous/NewInEFCore8/RawSqlSample.cs?name=SqlQueryAllColumns)]
+[!code-csharp[ComplexTypeConfig](../../../../samples/core/Miscellaneous/NewInEFCore8/ComplexTypesSample.cs?name=ComplexTypeConfig)]
 
-This query is parameterized and executed as:
+
+### Mutability
+
+In the example above, we ended up with the same `Address` instance used in three places. This is allowed and doesn't cause any issues for EF Core when using complex types. However, sharing instances of the same reference type means that if a property value on the instance is modified, then that change will be reflected in all three usages. For example, following on from above, let's change `Line1` of the customer address and save the changes:
+
+<!--
+        #region ChangeSharedAddress
+        customer.Address.Line1 = "Peacock Lodge";
+        await context.SaveChangesAsync();
+
+        context.ChangeTracker.Clear();
+-->
+[!code-csharp[ChangeSharedAddress](../../../../samples/core/Miscellaneous/NewInEFCore8/ComplexTypesSample.cs?name=ChangeSharedAddress)]
+
+This results in the following update to the database when using SQL Server:
 
 ```sql
-SELECT * FROM Posts as p WHERE p.PublishedOn >= @p0 AND p.PublishedOn < @p1
+UPDATE [Customers] SET [Address_Line1] = @p0
+OUTPUT 1
+WHERE [Id] = @p1;
+UPDATE [Orders] SET [BillingAddress_Line1] = @p2, [ShippingAddress_Line1] = @p3
+OUTPUT 1
+WHERE [Id] = @p4;
 ```
 
-The type used for query results can contain common mapping constructs supported by EF Core, such as parameterized constructors and mapping attributes. For example:
+Notice that all three `Line1` columns have changed, since they are all sharing the same instance. This is usually not what we want.
 
-<!--
-    public class BlogPost
-    {
-        public BlogPost(string blogTitle, string content, DateOnly publishedOn)
-        {
-            BlogTitle = blogTitle;
-            Content = content;
-            PublishedOn = publishedOn;
-        }
+> [!TIP]
+> If order addresses should change automatically when the customer address changes, then consider mapping the address as an entity type. `Order` and `Customer` can then safely reference the same address instance (which is now identified by a key) via a navigation property.
 
-        public int Id { get; private set; }
+A good way to deal with issues like this it to make the type immutable. Indeed, this immutability often natural when a type is a good candidate for being a complex type. For example, it usually makes sense to supply a complex new `Address` object rather than to just mutate, say, the country while leaving the rest the same.
```

Comment:
```suggestion
A good way to deal with issues like this is to make the type immutable. Indeed, this immutability is often natural when a type is a good candidate for being a complex type. For example, it usually makes sense to supply a complex new `Address` object rather than to just mutate, say, the country while leaving the rest the same.
```